types/src/on_chain_config/consensus_config.rs

```diff
@@ -47,13 +51,22 @@ impl OnChainConsensusConfig {
     pub fn max_failed_authors_to_store(&self) -> usize {
         match &self {
             OnChainConsensusConfig::V1(config) => config.max_failed_authors_to_store,
+            OnChainConsensusConfig::V2(config) => config.max_failed_authors_to_store,
         }
     }
 
     // Type and configuration used for proposer election.
     pub fn proposer_election_type(&self) -> &ProposerElectionType {
         match &self {
             OnChainConsensusConfig::V1(config) => &config.proposer_election_type,
+            OnChainConsensusConfig::V2(config) => &config.proposer_election_type,
+        }
+    }
+
+    pub fn quorum_store_enabled(&self) -> bool {
+        match &self {
+            OnChainConsensusConfig::V1(_config) => false,
+            OnChainConsensusConfig::V2(_config) => true,
```

Comment:
So we cannot use ConsensusConfigV2 and add it there?

Reply:
We could, but as Igor pointed out it adds a lot of boilerplate repeated configs.